strax/context.py

```diff
@@ -857,7 +857,7 @@ def get_iter(self, run_id: str,
 
         try:
             with contextlib.ExitStack() as stack:
-                if start_time != 0 and end_time != float('inf'):
+                if progress_bar and (start_time != 0 and end_time != float('inf')):
```

Comment:
Sorry, just saw it, could we change this. Instead of adding progress_bar here, we move the definition of start_time and end_time from line 838 and 839 to line 830.